### PR TITLE
Update ClinVar import - master / e104

### PR DESCRIPTION
**Pull request by ima23**
- add lookup by canonical spdi
- check all alleles for snps
- print change to warn

### CODE REVIEW

File: scripts/import/import_clinvar_xml [hunk 1]
```diff
@@ -1046,19 +1050,73 @@ sub get_variant{
...
-  else {
-    warn "No ref or alt allele or location found\n" if $DEBUG == 1;
-    return undef;
+  } else {
+    warn "no ref or alt or chr or start, skipping lookup by location identifier\n";
   }
+
+  #try to match by SPDI: longest allele format
+  #returns undef for ref==alt SPDIs
+  my $result = get_variant_via_SPDI ($record, $alt_allele);
```
Comment (sarahhunt) [inline scripts/import/import_clinvar_xml:1062]: Did this work in your testing?

get_variant_via_SPDI returns an object, an array of objects and a scalar, but only the scalar is caught here. I get an error message:
ERROR: Can't call method "add_evidence_value" without a package or object reference at scripts/import/import_clinvar_xml line 753.

and it's calling  add_evidence_value on the alt allele string '-'

**Comment by sarahhunt** [comment]
Generally, it is easier to spot the error messages you don't expect if you send your print the messages you expect.

**Comment by ima23** [comment]
Thanks @sarahhunt for checking. I've updated the array return and added ref checking.
I'm not sure what you mean by the error messages so I've removed the print/warn edit from this PR. Plan to update logs after I understand.

> Generally, it is easier to spot the error messages you don't expect if you print the messages you expect.

File: scripts/import/import_clinvar_xml (hunk 1 above)
Comment (ima23) [reply scripts/import/import_clinvar_xml]: I haven't seen that error before. What xml did you test? I can have a look.

Comment (sarahhunt) [reply scripts/import/import_clinvar_xml]: I'm using RCV000001471 - I've put the XML in the ClinVar/testdata/ directory on nfs.

**Comment by sarahhunt** [comment]
Data problems may trigger warnings when the import is running. If the logging messages you expect are sent to standard error, it is harder to see these and react to problems.
Ideally, only send messages you need to act on/fix/rerun for to standard error.

File: scripts/import/import_clinvar_xml [hunk 2]
```diff
@@ -1046,19 +1050,72 @@ sub get_variant{
...
-  else {
-    warn "No ref or alt allele or location found\n" if $DEBUG == 1;
-    return undef;
+  } else {
+    warn "no ref or alt or chr or start, skipping lookup by location identifier\n";
   }
+
+  #try to match by SPDI: longest allele format
+  #returns undef for ref==alt SPDIs
+  my ($var, $var_feats) = get_variant_via_SPDI ($record);
```
Comment (sarahhunt) [inline scripts/import/import_clinvar_xml:1062]: Can we return the alt allele here too? (For e!105 if not e!104)

Comment (ima23) [reply scripts/import/import_clinvar_xml:1062]: The method does not change the alt allele. The alt stays the same as before from hgvs.

Comment (sarahhunt) [reply scripts/import/import_clinvar_xml:1062]: If the variant is a deletion, HGVS reports only 'del' so we don't have the actual allele. We can obtain this from SPDI (with care)

**Pull request merged**